internal/actions/runWorkflow.go

```diff
@@ -278,12 +282,44 @@ func finalize(inputs finalizeInputs) error {
 			}
 		}
 
+		// add merging branch registry tag
+		if err = addDirectModeBranchTagging(); err != nil {
+			logging.Debug("failed to tag registry images: %v", err)
+		}
+
 		inputs.Outputs["commit_hash"] = commitHash
 	}
 
 	return nil
 }
 
+func addDirectModeBranchTagging() error {
+	wf, err := configuration.GetWorkflowAndValidateLanguages(true)
+	if err != nil {
+		return err
+	}
+
+	branch := strings.TrimPrefix(os.Getenv("GITHUB_REF"), "refs/heads/")
```

Comment:
Are we meant to be using the new `SPEAKEASY_ACTIVE_BRANCH` or is that being added to be used elsewhere

Reply:
No at this point in direct mode we are actually merging to main. So we want to promote main.

SPEAKEASY_ACTIVE_BRANCH is used by CLI. So when generation runs and we tag namespace we are tagging with the PR branch rather than defaulting to main which as discussed is inaccurate.